- title: 初学python可以做什么
- author: spark
- slug:What-can-a-beginner-do-with-Python
- date: 2018-11-20
- category:量化股票
- tags: stock,python,基本面,现金流


媳妇在学习了python两个月后，认为python可以做的事情

### 分析股票数据，指导投资

#### 用tushare获取某公司交易行情数据，分析大单交易

In [7]:
import pandas as pd
import numpy as np
import tushare as ts
import datetime
import hvplot.pandas
#获取某股票某天的大单交易，并查看数据
mydf = ts.get_sina_dd('300113','2018-11-19')
mydf.head()

code  name      time  price  volume  preprice type
0  300113  顺网科技  15:00:03  15.44  142800     15.43   买盘
1  300113  顺网科技  14:56:30  15.42   71700     15.42   买盘
2  300113  顺网科技  13:53:51  15.31   45000     15.31   卖盘
3  300113  顺网科技  13:31:24  15.26   54600     15.25   买盘
4  300113  顺网科技  13:07:54  15.20  131100     15.26   卖盘

In [8]:
#绘制点状图，查看大单分布情况
mydf.hvplot(x='price', y='volume',kind='scatter',title='test')

:Scatter   [price]   (volume)

#### tushare获取乐视网(300104)基本面现金流数据

In [9]:
df = ts.get_cash_flow('300104').T.head(10)
df

0              1               2             3                4   \
报表日期      单位  一、经营活动产生的现金流量  销售商品、提供劳务收到的现金       收到的税费返还  收到的其他与经营活动有关的现金   
20180930   元            NaN   1315751551.51  118997517.25     156739552.81   
20180630   元            NaN    958568433.19             0     172393232.74   
20180331   元            NaN    377926049.86             0      64387459.62   
20171231   元            NaN   5453436883.59   20404575.71    1162302902.63   
20170930   元            NaN   4476235969.63   20404575.71     858171870.33   
20170630   元            NaN   3583676403.36   20404575.71     374225110.23   
20170331   元            NaN   2630848829.83    4935395.81     778797096.80   
20161231   元            NaN  14634188704.16   30288787.69     682734841.98   
20160930   元            NaN  11833407221.01     919803.41    1210361330.01   

                      5               6                7             8   \
报表日期          经营活动现金流入小计  购买商品、接受劳务支付的现金  支付给职工以及为职工支付的现金       支付的各项税费   
20180930   1610065753.09   1572239784.29     390822548.33   42774973.73   
20180630   1145916081.73   1369524082.60     269368739.96   24369547.79   
20180331    452972279.20    695489706.69     138366140.36   18417124.38   
20171231   6686190978.78   6883536068.86    1076404820.24  176840386.59   
20170930   5399875744.70   6400620812.77     749330501.59  165136004.59   
20170630   4019234054.09   4856930049.89     509291781.55  136215690.09   
20170331   3283611843.31   3226163987.21     261428176.31   66178271.24   
20161231  15777018767.95  12721979030.65    1098193418.77  404567218.99   
20160930  13113995720.83   9859277182.42     756657743.04  341692556.96   

                       9        ...                  66              67  \
报表日期      支付的其他与经营活动有关的现金       ...                  其他    经营活动产生现金流量净额   
20180930     524708311.67       ...                   0               0   
20180630     292728150.75       ...                   0   -852700993.05   
20180331     237108495.46       ...                   0               0   
20171231    1166220140.52       ...        -66428110.61  -2640545383.24   
20170930     858881227.29       ...                   0               0   
20170630     662557068.69       ...        -34293119.37  -2185441404.02   
20170331     590961188.51       ...                   0               0   
20161231    1889108245.05       ...        135729333.71  -1068060768.96   
20160930    1038054897.72       ...                   0               0   

              68             69            70             71             72  \
报表日期      债务转为资本  一年内到期的可转换公司债券      融资租入固定资产        现金的期末余额        现金的期初余额   
20180930       0              0             0              0              0   
20180630       0              0             0   469608596.69   725439935.91   
20180331       0              0             0              0              0   
20171231       0              0             0   725439935.91  1469207975.00   
20170930       0              0             0              0              0   
20170630       0              0             0   802146634.86  1469207975.00   
20170331       0              0             0              0              0   
20161231       0           0.00  257630871.91  1469207975.01  2714778115.15   
20160930       0              0             0              0              0   

                  73          74              75  
报表日期      现金等价物的期末余额  现金等价物的期初余额   现金及现金等价物的净增加额  
20180930           0           0               0  
20180630           0           0   -255831339.22  
20180331           0           0               0  
20171231           0           0   -743768039.09  
20170930           0           0               0  
20170630           0           0   -667061340.14  
20170331           0           0               0  
20161231        0.00        0.00  -1245570140.14  
20160930           0           0               0  

[10 rows x 76 columns]

从现金流上可以看出，20161231开始，公司基本面出现重大问题

### 爬虫抓取特定兴趣新闻，专注高效

1. 财经社新闻抓取，获取特定关键词新闻
2. 发消息到自己钉钉

### 碎片化中文数据挖掘

1. 提取用户搜索日志
2. 挖掘用户关注的热点